app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/LoadingActivity.java

```diff
@@ -1,21 +1,108 @@
 package edu.drexel.lapcounter.lapcounter.frontend;
 
+import android.arch.lifecycle.ViewModelProviders;
 import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.util.Log;
 import android.view.View;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import edu.drexel.lapcounter.lapcounter.R;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Device.DeviceRepository;
+import edu.drexel.lapcounter.lapcounter.backend.Database.State.State;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Transition.TransitionRepository;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Units.Units;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.Workout;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.WorkoutRepository;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.WorkoutViewModel;
+import edu.drexel.lapcounter.lapcounter.backend.TimestampConverter;
 
 public class LoadingActivity extends AppCompatActivity {
 
+    /*** final static strings for units ***/
+    public final static String unitsMeters = "Meters";
+    public final static String unitsYards = "Yards";
+
+    /*** final static string for states ***/
+    public final static String statesNear = "NEAR";
+    public final static String statesFar = "FAR";
+    public final static String statesUnknown = "UNKNOWN";
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_loading_screen);
 
         // In the final version, use R.string.<string id> for titles
         getSupportActionBar().setTitle("Splash Screen");
+        WorkoutViewModel mWorkoutViewModel =  ViewModelProviders.of(this).get(WorkoutViewModel.class);
+
+
+        try
+        {
+            //Units setup
+            List<Units> units = new ArrayList<>();
```

Comment:
I think this should be wrapped up in a function in the database code, maybe a method at the repository level (I don't think there's a need to do this at the ViewModel level)